### DISCUSSION TITLE
Was the Shift button meant for something greater.....

### DISCUSSION THREAD

**Discussion by Electribe-X** [bangcorrupt/hacktribe#192] [General]
Dear bangcorrupt,

First of all well done on this, absolutely fantastic work. As a long time Electribe 2 user (have both variations), I've always felt that the device has yet to reach it's true potential, and the fact that you guys have the skills and knowledge to be able to actually work towards this, is awesome. I've installed it and have had a lot of fun so far.

If you have a moment I'd love to get your thoughts on something I started thinking about not long after I started using the devices.

I have always suspected the original plan for the OS had a lot more features included, relating to the use of the Shift button, but for whatever reasons, Korg did not implement them.

For example, I'm pretty sure that the Insert FX were originally envisioned to have control for 2 parameters, like it's predecessor had. I believe the plan was, you would have adjusted the 2nd parameter by holding down the Shift button, before adjusting the IFX Edit pot.

Another example, it may have been possible to hold the Shift button, and then adjusting the Volume pot would adjust the level of the signal of that particular pad being sent to the Master FX.

Or just to go super mad scientist for a moment, there was supposed to be 2 Modulation generators. You would hold Shift, and then all the pots in the Modulation section, would then adjust the parameters of Modulation generator 2.

Would love to know if you have ever seen any evidence that this may have been the case, and just hypothetically speaking, would it actually be technically possible to implement on the devices?

Regards,
Dave

**Comment by bangcorrupt** [comment]
The IFX knob is a matrix control, mapped to multiple parameters at once with different scales and ranges.  Editing of each individual parameter is on the way, this will allow setting up new presets and control mappings. 

I know how most of the controls are processed now, so in time I'll be able to set up modifier keys for all sorts of things.  I'd like to make the XY pad freely assignable too.

I'm not sure if I'll be able to add extra LFO's, but hopefully I can turn velocity and key tracking into modulation sources.  The envelope may be hiding some complexity, I think the Punch IFX is part of the envelope and we may be able to have a full ADSR.

The only thing I've seen that looks like it was deactivated at a late stage is 8 notes per step.  There is space for them in the pattern file and a lot of the code that processes steps loops through 8 iterations, but somewhere it is limited to 4.

It's essentially the KingKorg sound engine multiplied by 16 with a lot of the polyphony and complexity cut down (I think they have the same processor).  I'm not sure how much of the original sound engine is compiled into the binary, but I'll gradually expose what's there.

**Comment by listenForward** [comment]
Hey BangCorrupt, 
Great to hear that you're working to expand options and fluency of the modifier keys.

For as much as the SHIFT key is vital, I've always been annoyed that ... 
In terms of Ergonomic Affordances:
~ not special or ergonomically distinct, the same size as other keys
~ right in the middle of other buttons, easy to miss with the consequences of hitting other things
~ holding SHIFT to press other buttons is not easy to reach all 16 pads with one hand, so the Pads on the Right side REQUIRE 2 handed gestures.
~ only momentary, not imbued with any option to latch, so you ALWAYS have to HOLD the SHIFT button...

and the SHIFT key's Current Mapping 
~ with the 16 Pads to jump to specific pages in the Pattern/Part/Step/PartUtil/Global Menu tree... with UI from the Encoder
~ Osc/Samlpe Knob to jump Categories
~ Modulation Knob to jump DifferntSource@SameDestinationParameter
~ Insert FX knob +/-  by 10 instead of 1. 

Given the the (present) Affordances and Mapping above, I find the SHIFT  key behavior is more "editorial" than "performative", and I notice that I tend to use the Shift function more when the unit is stopped than during run/play, 
Perhaps we should respect this, especially since since the SHIFT key remains crowded up by the screen and encoder.

So, if/when you'd considering expanding the Macros of the SHIFT (and other keys) would you consider implementing some HOLD mode for SHIFT ?  
1)  "Double Tap" and/or "Hold > 2 Sec" to enter ShiftHeld mode,
2) tap SHIFT-enabled buttons to move between successive short-cuts 
3) single tap Shift to exit the ShiftHeld mode.

I know there's no LED under the SHIFT key, but perhaps we can have some Momentary/Held UI of the Shift state on the main screen ?

Cheers

**Comment by listenForward** [comment]
And since I forgot to work it in, I'd just want to (re)state that I'm excited for additional Modifier Keys and shortcuts... but I'd hope these would be designed
~ Away from the SHIFT key, given the concerns above
~ toward more semantic combinations, especially w/ Pad Modes
~ to incorporate more contextually explicit usage of the BAR Keys 1/2/3/4 and/or other Mode keys

Here's some brainstorms/suggestions of 2 examples  "semantic combination"

== for STEP JUMP ==

At present for Step Jump mode, we...
0) note the 1/2/3/4 Bar/Page keys light Blue for PlayPosition and Red for EditView, which IS relevant for this Sequencer UI.
1) enter Step Jump mode by pressing the STEP JUMP button
2) use the 1/2/3/4 keys to select among  4 Bars/Pages
3) push one of the 16 Pads for the Track  we wish to Jump/ReAlign to Offset
 
...Imagine adding:
4) hold the STEP JUMP button and press 1/2/3/4 to Jump directly to the first step of that bar, achieving in 1 action with one hand what normally took 3+ pokes.
5) hold STEP JUMP and pres TAP to reset Current part to it's original StepCount ( as if it hadn't been Jumped/ReAligned). Perhaps this would require some reference of a Main/Master step-counter.
6) hold STEP JUMP and press PART< or PART> to align the Current Part to the StepCount/Offset of the <<Previous or >>Next part ...
7) hold STEP JUMP and press the PLAY/PAUSE key to reset ALL track step-offset to it's original value, so you don't have to STOP and RESTART by hand...


== PART MUTE with MuteTiming and Scenes ===

For the Mute Mode, we...
0) note, the buttons 1/2/3/4 still light Blue to show current Bar/Page of play, and Red to indicate which Bar/Page the Sequencer would edit (but this is Irrelevant w/in Mute View ...)
1) press PART MUTE to enter Part Mute mode (solid LED)
2) use pads 1-16 to toggle mute for tracks 1-16
3) hold PART MUTE button to enter SOLO mode (flashing LED)...where next pad 1-16 solo-s that track (muting all others), which falls immediately back to Mute Mode (solid LED) ...you'd have to manually UN-mute other tracks, w/ no memory of mixed track muting from before.

...imagine adding:
4) Quantization of Mute Actions so you could set up multiple things to un/mute together w/out having to "play chords" on the pad. While in Mute Mode, the 1/2/3/4 Bar Keys can be used pushed (w/ Red lighting) adjust the Quantization of un/mute actions (using Red LED to show options between Instant/Beat/Bar/MasterPatternStart).
5.0) up to 4x Mute Scenes.... 
5.1 )... recalled from the (flashing) Solo state the MUTE button, the 1/2/3/4 (bar) keys flash (red) as if/where there is some MuteScene stored there, and tap the 1/2/3/4 keys to recall them 
5.2) ...stored by having to hold the MUTE button and then hold the 1/2/3/4 BAR key to store.


Hope this helps w/ other understanding/interest/ideas for further UX extensions beyond the SHIFT key.
